### ISSUE TITLE
Uncaught interrupted_exc_t in log_outdated_indexes

### ISSUE THREAD

**Issue by Tryneus** [rethinkdb/rethinkdb#4977]
When shutting down a server after the table_fuzzer, experienced the following crash:

```cpp
error: Error in ../src/errors.cc at line 125:
error: Uncaught exception of type "interrupted_exc_t"
  what(): interrupted
error: Backtrace:
error: Wed Oct 21 17:28:51 2015

1 [0x1bb88ac]: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:101
2 [0x17dd771]: backtrace_t::backtrace_t() at backtrace.cc:203
3 [0x17ddddf]: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
4 [0x17dd6bf]: format_backtrace(bool) at backtrace.cc:198
5 [0x17d15a2]: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
6 [0x17d1850]: terminate_handler() at errors.cc:125
7 [0x7fdde6eecd46]: /opt/gcc-4.9.2/lib64/libstdc++.so.6(+0x63d46) [0x7fdde6eecd46] at 0x7fdde6eecd46 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
8 [0x7fdde6eecd91]: /opt/gcc-4.9.2/lib64/libstdc++.so.6(+0x63d91) [0x7fdde6eecd91] at 0x7fdde6eecd91 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
9 [0x7fdde6eecfa8]: /opt/gcc-4.9.2/lib64/libstdc++.so.6(+0x63fa8) [0x7fdde6eecfa8] at 0x7fdde6eecfa8 (/opt/gcc-4.9.2/lib64/libstdc++.so.6)
10 [0x16cc7d7]: wait_interruptible(signal_t const*, signal_t const*) at interruptor.cc:11
11 [0x1bceb4a]: /home/ssd2/grey/rethinkdb/build/debug/rethinkdb() [0x1bceb4a] at 0x1bceb4a ()
12 [0x1bce56c]: outdated_index_issue_tracker_t::log_outdated_indexes(multi_table_manager_t*, cluster_semilattice_metadata_t const&, signal_t*) at outdated_index.cc:112
13 [0x1bdf3d7]: do_serve(io_backender_t*, bool, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:345
14 [0x1be16a3]: serve(io_backender_t*, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:641
15 [0x1c0c467]: run_rethinkdb_serve(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, uuid_u const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*) at command_line.cc:807
16 [0x1c21b4a]: void std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, uuid_u*, server_config_versioned_t*, cluster_semilattice_metadata_t*, directory_lock_t*, bool*))(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, uuid_u const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*)>::__call<void, , 0, 1, 2, 3, 4, 5, 6, 7, 8, 9>(std::tuple<>&&, std::_Index_tuple<0, 1, 2, 3, 4, 5, 6, 7, 8, 9>) at functional:1147
17 [0x1c1f928]: void std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, uuid_u*, server_config_versioned_t*, cluster_semilattice_metadata_t*, directory_lock_t*, bool*))(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, uuid_u const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*)>::operator()<, void>() at functional:1206
18 [0x1c1c7fc]: std::_Function_handler<void (), std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> >, uuid_u*, server_config_versioned_t*, cluster_semilattice_metadata_t*, directory_lock_t*, bool*))(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optional<boost::optional<unsigned long> > const&, uuid_u const*, server_config_versioned_t const*, cluster_semilattice_metadata_t const*, directory_lock_t*, bool*)> >::_M_invoke(std::_Any_data const&) at functional:1780
19 [0x15ee45c]: std::function<void ()>::operator()() const at functional:2162
20 [0x16021d6]: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
21 [0x1602ffa]: std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()(starter_t*, std::function<void ()> const&) const at functional:551
22 [0x1602d85]: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::__call<void, , 0, 1>(std::tuple<>&&, std::_Index_tuple<0, 1>) at functional:1147
23 [0x16028dc]: void std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)>::operator()<, void>() at functional:1206
24 [0x16025da]: std::_Function_handler<void (), std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)> >::_M_invoke(std::_Any_data const&) at functional:1780
25 [0x15ee45c]: std::function<void ()>::operator()() const at functional:2162
26 [0x1603304]: callable_action_instance_t<std::function<void ()>&>::run_action() at callable_action.hpp:31
27 [0x15ffa04]: callable_action_wrapper_t::run() at runtime_utils.cc:43
28 [0x1607c0b]: coro_t::run() at coroutines.cc:207
29 [0x1602467]: coro_t* coro_t::spawn_sometime<std::function<void ()>&>(std::function<void ()>&) at coroutines.hpp:58
30 [0x16021b3]: starter_t::on_thread_switch() at runtime.cc:57
31 [0x15feea3]: linux_message_hub_t::on_event(int) at message_hub.cc:154
32 [0x15f7f3f]: epoll_event_queue_t::run() at epoll.cc:115
33 [0x1603cda]: linux_thread_pool_t::start_thread(void*) at thread_pool.cc:169
```

This is probably not high priority because the server was shutting down anyway.

**Comment by Tryneus** [comment]
Looks like `log_outdated_indexes` should just catch and ignore `interrupted_exc_t`s, since `visit_tables` can be interrupted.

**Comment by Tryneus** [comment]
This should probably be handled in the other place that calls `visit_tables`: `jobs_manager_t::on_get_job_reports`.

**Comment by Tryneus** [comment]
Fix is up in review 3292.

**Comment by Tryneus** [comment]
The fix was approved and merged to `next` and `v2.1.x` in commits 990e757 and aba3283, respectively.  Will be in release 2.1.6.
